apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoBar.tsx

```diff
@@ -78,20 +78,22 @@ const CourseInfoBar = (props: CourseInfoBarProps) => {
                     const courseId = encodeURIComponent(
                         `${deptCode.replace(/\s/g, '')}${courseNumber.replace(/\s/g, '')}`
                     );
-                    const response = await fetch(`${PETERPORTAL_REST_ENDPOINT}/courses/${courseId}`);
+                    const res: RawResponse<Course> = await fetch(
+                        `${PETERPORTAL_REST_ENDPOINT}/courses/${courseId}`
+                    ).then((r) => r.json());
+
+                    if (!isErrorResponse(res)) {
+                        const data = res.payload;
 
-                    if (response.ok) {
-                        const jsonResp = (await response.json()) as CourseResponse;
-                        console.log(jsonResp.prerequisite_list);
                         setCourseInfo({
-                            id: jsonResp.id,
-                            title: jsonResp.title,
-                            prerequisite_tree: jsonResp.prerequisite_tree,
-                            prerequisite_list: jsonResp.prerequisite_list,
-                            prerequisite_text: jsonResp.prerequisite_text,
-                            prerequisite_for: jsonResp.prerequisite_for,
-                            description: jsonResp.description,
-                            ge_list: jsonResp.ge_list.join(', '),
+                            id: data.id,
+                            title: data.title,
+                            prerequisite_tree: data.prerequisiteTree,
```

Comment:
Sike, doesn't look good to me. The migration breaks this part because `data.prerequisiteTree` used to be a string-serialized JSON object but now it's a real object. @teresa-liang or @Voark can you fix this?

Reply:
This issue has been resolved in commit f7c41696766521aac31fb36076d7165c75513800!